samples/layered-java-application/build.gradle

```diff
@@ -0,0 +1,85 @@
+ *
+ * The Universal Permissive License (UPL), Version 1.0
+ *
+ * Subject to the condition set forth below, permission is hereby granted to any
+ * person obtaining a copy of this software, associated documentation and/or
+ * data (collectively the "Software"), free of charge and under any and all
+ * copyright rights in the Software, and any and all patent rights owned or
+ * freely licensable by each licensor hereunder covering either (i) the
+ * unmodified Software as contributed to or provided by such licensor, or (ii)
+ * the Larger Works (as defined below), to deal in both
+ *
+ * (a) the Software, and
+ *
+ * (b) any piece of software and/or hardware listed in the lrgrwrks.txt file if
+ * one is included with the Software each a "Larger Work" to which the Software
+ * is contributed by such licensors),
+ *
+ * without restriction, including without limitation the rights to copy, create
+ * derivative works of, display, perform, and distribute the Software and make,
+ * use, sell, offer for sale, import, export, have made, and have sold the
+ * Software and the Larger Work(s), and to sublicense the foregoing rights on
+ * either these or other terms.
+ *
+ * This license is subject to the following condition:
+ *
+ * The above copyright notice and either this complete permission notice or at a
+ * minimum a reference to the UPL must be included in all copies or substantial
+ * portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+plugins {
+    id 'application'
+    id 'org.graalvm.buildtools.native'
+}
+
+repositories {
+    mavenCentral()
+}
+
+application {
+    mainClass.set('org.graalvm.demo.Application')
+}
+
+def junitVersion = providers.gradleProperty('junit.jupiter.version')
+        .get()
+
+dependencies {
+    implementation("org.slf4j:slf4j-api:2.0.17")
+    runtimeOnly("ch.qos.logback:logback-classic:1.5.18")
+    testImplementation(platform("org.junit:junit-bom:${junitVersion}"))
+    testImplementation('org.junit.jupiter:junit-jupiter')
+}
+
+test {
+    useJUnitPlatform()
+}
+
+tasks.named("nativeRun") {
+    runtimeArgs.add(providers.gradleProperty("message").orElse("default message"))
+}
+
+graalvmNative {
+    binaries {
+        libdependencies {
+            createLayer {
+                modules = ["java.base"]
+                jars.from(externalDependenciesOf(configurations.runtimeClasspath))
+            }
+        }
+        main {
+            useLayer("libdependencies")
+        }
+    }
```

Comment:
Can you please also add a configuration that could switch off the layered build? In your early demo if I recall correctly you had an `-incremental` flag. It would be useful to be able to easily switch between these 3 scenarios (and maybe other in the future):
 - a regular build (no layers)
 - a layered build that has just `java.base` in the base layer
 - a layered build that has `java.base` and all dependencies in the base layer

Reply:
Again it's unclear to me what you want. These are tests of the plugin itself, it doesn't quite make sense to add these options, since they wouldn't be used by our tests.

Reply:
Moreover, for demo purposes, it would be nice if we could run the scenarios without the functional tests infrastructure, as standalone apps. Can you please add some minimal documentation with the bare bone commands one needs to do that?

Reply:
I see, then what would be the best way to showcase this plugin functionality to an end user?

Reply:
Maybe after we merge the plugin changes we can add a demo project to the Micronaut repo?

Reply:
Yes. Although once this is merged and NBT released, I could implement this directly in the Micronaut Gradle plugins, where the options would make more sense, and the configuration could be hidden.